src/main/java/requestManagement/fleetManager/storage/HostDAOFileSystem.java

```
@@ -0,0 +1,157 @@
+package requestManagement.fleetManager.storage;
+
+import requestManagement.fleetManager.hosts.Host;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class HostDAOFileSystem implements HostDAO{
+
+    private String filename= "Hosts.txt";
+    private FileWriter fw;
+    private FileReader fr;
+    private List<List<String>> dataFromFile;
+    private List<String> temp;
+
+    public HostDAOFileSystem(){
+
+    }
+
+    @Override
+    public void addHost(Host host){
+        readFile();
+        if(!contains(host.getIpv4())) {
+            try {
+                fw = new FileWriter(filename, true); //the true will append the new data
+                fw.write("\n" + host.getIpv4() + "," + host.getDNS() + "," + host.getPort() + "," + host.getState() + "," + host.getSubnet());//appends the string to the file
+                fw.close();
+            } catch (IOException ioe) {
+                System.err.println("IOException: " + ioe.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void removeHost(Host host) {
+        readFile();
+        try {
+            if (contains(host.getIpv4())) {
+                for (int i = 0; i < dataFromFile.size(); i++) {
+                    if (dataFromFile.get(i).contains(host.getIpv4())) {
+                        dataFromFile.remove(i);
+                    }
+                }
+                writeToFile();
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+
```

Empty catch block. Have some sort of error printing out

---

Done